ui/app/store/actions.js

```diff
@@ -2304,57 +2037,6 @@ function pairUpdate (coin) {
   }
 }
 
-function shapeShiftSubview () {
-  const pair = 'btc_eth'
-  return (dispatch) => {
-    dispatch(actions.showSubLoadingIndication())
-    shapeShiftRequest('marketinfo', { pair }, (mktResponse) => {
-      shapeShiftRequest('getcoins', {}, (response) => {
-        dispatch(actions.hideSubLoadingIndication())
-        if (mktResponse.error) {
-          return dispatch(actions.displayWarning(mktResponse.error))
-        }
-        dispatch({
-          type: actions.SHAPESHIFT_SUBVIEW,
-          value: {
-            marketinfo: mktResponse,
-            coinOptions: response,
-          },
-        })
-      })
-    })
-  }
-}
-
-function coinShiftRquest (data, marketData) {
-  return (dispatch) => {
-    dispatch(actions.showLoadingIndication())
-    shapeShiftRequest('shift', { method: 'POST', data }, (response) => {
-      dispatch(actions.hideLoadingIndication())
-      if (response.error) {
-        return dispatch(actions.displayWarning(response.error))
-      }
-      const message = `
-        Deposit your ${response.depositType} to the address below:`
-      log.debug(`background.createShapeShiftTx`)
-      background.createShapeShiftTx(response.deposit, response.depositType)
-      dispatch(actions.showQrView(response.deposit, [message].concat(marketData)))
-    })
-  }
-}
-
-function buyWithShapeShift (data) {
-  return () => new Promise((resolve, reject) => {
-    shapeShiftRequest('shift', { method: 'POST', data }, (response) => {
-      if (response.error) {
-        return reject(response.error)
-      }
-      background.createShapeShiftTx(response.deposit, response.depositType)
```

Comment:
Now that this has been removed, the background method `createShapeShiftTx` isn't called anywhere anymore. It can be removed from `metamask-controller` (though the controller is still read from, so that has to stay for now).

Reply:
Updated